pkg/controller/bundle/bundle_unpacker.go

```diff
@@ -65,7 +65,7 @@ func newBundleUnpackResult(lookup *operatorsv1alpha1.BundleLookup) *BundleUnpack
 	}
 }
 
-func (c *ConfigMapUnpacker) job(cmRef *corev1.ObjectReference, bundlePath string) *batchv1.Job {
+func (c *ConfigMapUnpacker) job(cmRef *corev1.ObjectReference, bundlePath string, secrets []corev1.LocalObjectReference) *batchv1.Job {
```

Comment:
This looks good, but do we want to go with the same approach of using the SA instead of passing the secret directly so that the `imagePullSecrets` field on the pod spec would get set automatically?

Or are we unable to get the catalog SA since it's generated randomly. 

Maybe its better to use the same SA for pulling an index and unpacking a bundle from that same index via a map of some kind

Reply:
@exdx the issue with that approach is that the serviceaccount is created when the registry pod is created, but no record of the serviceaccount is persisted in the CatalogSource itself (it does not have any `spec.Serviceaccount` field. The SA is attached to the registry pod after both are created). We could query for the registry pod of the CatalogSource and extract the SA name from there, but that seems like an unnecessary extra client call when the CatalogSource is already being fetched once.

Reply:
We could get around this by annotating the catalogsource when creating it with the service account that is being used by the underlying pod and then using it in this context. I just feel like if we went through this effort of getting per-catalog SAs shouldn't we use them for the underlying bundles as well? Seems like a clean solution but would like @njhale to weigh in here